src/richie/apps/courses/models/organization.py

```diff
@@ -167,6 +168,29 @@ def get_persons(self, language=None):
         """
         return self.get_reverse_related_page_extensions("person", language=language)
 
+    @staticmethod
+    def get_organizations_codes(page, language: str):
+        """
+        Return organization attached to the page and
+        organizations linked to the current page via an organization plugin in any of the
+        placeholders on the page
+        """
+        return (
+            Organization.objects.filter(
+                Q(
+                    extended_object__organization_plugins__cmsplugin_ptr__language=language,
+                    extended_object__organization_plugins__cmsplugin_ptr__placeholder__page=page,
+                )
+                | Q(
+                    extended_object__exact=page,
+                ),
+                extended_object__title_set__published=True,
+                code__isnull=False,
+            )
+            .distinct()
+            .values_list("code", flat=True)
+        )
+
```

Comment:
You have to write a test for this method.
Furthermore, we are aware about db queries performance issues so as this method is called on every page, please ensure that once a page is cached, db queries related to this method are also cached.

Reply:
Ok. I have added a test for that method.

> please ensure that once a page is cached, db queries related to this method are also cached.

I've cached the `Organization.get_organizations_codes` method directly on the organization.py file.
Alternatively I could replace with a lazy evalution of the queryset on the django `base.html` template. I don't know django good enough to know if this alternative would be better.